kernel/kernel-api/src/main/java/io/delta/kernel/types/StructType.java

```diff
@@ -93,7 +95,14 @@ public int indexOf(String fieldName) {
     return fieldAndOrdinal != null ? fieldAndOrdinal._2 : -1;
   }
 
+  /**
+   * @return the StructField with the given name
+   * @throws IllegalArgumentException if the field does not exist
+   */
   public StructField get(String fieldName) {
```

Comment:
I agree this is a strict improvement over NPE, but do we now expect every call to `StructField.get` to wrap it with a try and catch?

Could we also, in this PR, at least add a `getOpt` that returns an `Optional<String>` and this could give some more flexibility to callers?